app/controllers/mixins/automation_mixin.rb

```diff
@@ -0,0 +1,42 @@
+module Mixins
+  module AutomationMixin
+    # Returns an object to be reused for passing down to service catalogs(rails) and dialog(angular) form.
+    # used in catalogs_helper, miq_ae_customization_helper, catalog_controller, editor.html.haml
+    AUTOMATION_TYPES = {
+      :automate => {
+        :key    => 'embedded_automate',
+        :label  => _('Embedded Automate'),
+        :fields => ['ae_namespace', 'ae_class', 'ae_instance', 'ae_message', 'ae_attributes'],
+      },
+      :workflow => {
+        :key    => 'embedded_workflow',
+        :label  => _('Embedded Workflow'),
+        :fields => ['configuration_script_id', 'workflow_name'],
+      },
+    }
+
+    # Returns an array of entry point options.
+    # used in provision_entry_points, retirement_entry_points, reconfigure_entry_points html files
+    def automation_type_options
+      options = [["<#{_('No Entry Point')}>", nil]]
+      options + AUTOMATION_TYPES.map do |item|
+        [item[1][:label], item[1][:key]]
+      end
```

Comment:
This looks like a weird way to dereference a hash (the item[1] is actually just the value where item[0] is the hash key)

```suggestion
      options + AUTOMATION_TYPES.values.map do |item|
        [item[:label], item[:key]]
      end
```